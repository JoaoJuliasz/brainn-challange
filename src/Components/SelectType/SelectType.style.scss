.lottery-background {
    height: 100%;
    position: relative;

    &-0 {
        background: #6befa3;
        height: calc(100% - 20px);
        padding: 10px;
    }
    &-1 {
        background: #8666ef;
        height: calc(100% - 20px);
        padding: 10px;
    }
    &-2 {
        background: #dd7ac6;
        height: calc(100% - 20px);
        padding: 10px;
    }
    &-3 {
        background: #ffab64;
        height: calc(100% - 20px);
        padding: 10px;
    }
    &-4 {
        background: #5aad7d;
        height: calc(100% - 20px);
        padding: 10px;
    }
    &-5 {
        background: #bfaf83;
        height: calc(100% - 20px);
        padding: 10px;
    }

    &:before {
        width: 210px;
        display: inline-block;
        vertical-align: middle;
        margin-right: 10px;
        content: "";
        height: 100%;
        position: absolute;
        right: -110px;
        border-radius: 50% 0 0 50%;
        background: rgb(233, 233, 233);

        @media (max-width: 860px) {
            bottom: -30px;
            width: 100%;
            left: 0;
            height: 95px;
            border-radius: 50% 50% 0 0;
        }
    }

    .select-container {
        height: 10%;
        display: flex;
        align-items: center;
    }

    .contest-info-container {
        height: 90%;
        display: flex;
        color: #fff;
        position: relative;

        .lottery-info {
            display: flex;
            align-items: center;
            img {
                max-width: 65px;
                max-height: 65px;
            }

            p {
                font-size: 27px;
                margin: 10px;
                text-transform: capitalize;
                font-weight: 700;
            }
        }

        .contest-info {
            position: absolute;
            bottom: 20px;
            // width: 100%;
            text-align: left;
            right: 0;
            left: 12%;
            font-size: 20px;
            font-weight: 500;
        }

        @media(max-width: 860px) {
            height: 60%;

            .contest-info {
                bottom: -30px;
            }
        }

        @media(min-width: 861px) and (max-width: 1050px) {
            .lottery-info {
                width: 58%;

                p {
                    margin: 2px;
                    font-size: 25px;
                }
            }
        }
    }
}
