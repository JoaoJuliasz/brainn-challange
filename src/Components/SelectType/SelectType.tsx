import React, { useEffect } from 'react';
// @ts-ignore
import instance from '../../config/config'
// @ts-ignore
import logo from '../../assets/Logo_Sena.png'
import Select, { StylesConfig } from 'react-select'
// @ts-ignore
import { ValueType } from "react-select/lib/types";
import './SelectType.style.scss'
import { Lotteries, Contest, LotteriesContests } from '../dataTypes/types'

type SelectLotteryProps = {
    lotteries: Lotteries[],
    selectedLottery: string,
    setSelectedLottery: (value: string) => void,
    lotteriesContests: LotteriesContests,
    contest: Contest,
    setContest: (value: Contest) => void,
    setLoading: (value: boolean) => void
}

const SelectType = ({ lotteries, selectedLottery, setSelectedLottery, lotteriesContests, contest, setContest, setLoading }: SelectLotteryProps) => {

    const selectStyles: StylesConfig = {
        container: () => ({ width: 200, margin: '20px' }),
        menu: () => ({ top: 'auto', background: '#fff', position: 'absolute', width: 200, borderRadius: '10px', margin: '5px 0', zIndex: 3 }),
        indicatorSeparator: () => ({ display: 'none' }),
        control: () => ({
            border: 'none', borderRadius: '10px', boxShadow: '5px 5px 31px -10px #000000',
            display: 'flex', flexWrap: 'wrap', background: '#fff', justifyContent: 'space-around', minHeight: '38px', transition: 'all 100ms'
        })
    }


    const getContest = (id: number) => {
        setLoading(true)
        const contestId = lotteriesContests?.find(contest => contest.loteriaId === id)
        instance.get<Contest[]>(`concursos/${contestId?.concursoId}`)
            .then((res: any) => {
                setContest(res.data)
                setLoading(false)
            })
    }

    const formatDate = (item: string) => {
        let date = new Date(item),
            day = date.getDate().toString(),
            dayF = (day.length === 1) ? '0' + day : day,
            month = (date.getMonth() + 1).toString(),
            monthF = (month.length === 1) ? '0' + month : month,
            yearF = date.getFullYear()
        return dayF + "/" + monthF + "/" + yearF;
    }

    const selectLotteryChange = (option: ValueType<Lotteries>) => {
        setSelectedLottery(option.value)
        getContest(option.value)
    }

    useEffect(() => {
        getContest(0)
    }, [lotteriesContests])

    return (
        <div className="lottery-background">
            <div className={`lottery-background-${selectedLottery}`}>
                <div className="select-container">
                    <Select
                        onChange={selectLotteryChange}
                        value={lotteries[parseInt(selectedLottery)]}
                        options={lotteries}
                        styles={selectStyles}
                    />
                </div>
                {contest &&
                    <div className="contest-info-container">
                        <div className="lottery-info">
                            <img src={logo} alt="logo-loteria" />
                            <p>{lotteries[contest.loteria]?.label}</p>
                        </div>

                        <div className="contest-info">
                            <p>Concurso</p>
                            <p>{contest.id} - {contest.data ? formatDate(contest.data) : ''}</p>
                        </div>
                    </div>}
            </div>
        </div>
    );
};

export default SelectType;