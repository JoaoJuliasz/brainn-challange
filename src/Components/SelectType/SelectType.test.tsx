import React from 'react'
// @ts-ignore
import { shallow } from 'enzyme'
import SelectType from './SelectType'

describe('SelectType', () => {
    const mockContest = {
        id: '456',
        loteria: 6,
        numeros: ['1', '4', '5'],
        data: ''
    }

    const mockLotteries = [{
        value: 1,
        label: 'mega-sena'
    },
    {
        value: 2,
        label: 'quina'
    }
    ]

    const mockSelectedLottery = '1'

    const mockLoterriesContests = [{
        loteriaId: 6,
        concursoId: '456'
    }]

    const mockSetSelectedLottery = jest.fn();
    const mockSetLoading = jest.fn();
    const mockSetContest = jest.fn();

    const wrapper = shallow(<SelectType contest={mockContest} lotteries={mockLotteries} selectedLottery={mockSelectedLottery} lotteriesContests={mockLoterriesContests} setSelectedLottery={mockSetSelectedLottery} setLoading={mockSetLoading} setContest={mockSetContest} />)
    it('expect to render SelectType component', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should change lottery value', () => {
        wrapper.find('[id="lottery-selector"]').simulate('change', { target: { value: { value: 2, label: 'quina' } } })
        expect(wrapper.find('[id="lottery-selector"]').props().value).toStrictEqual({ value: 2, label: 'quina' })
        expect(mockSetSelectedLottery).toHaveBeenCalled()
    })
})