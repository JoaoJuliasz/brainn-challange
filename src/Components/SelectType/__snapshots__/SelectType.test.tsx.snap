// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectType expect to render SelectType component 1`] = `
<div
  className="lottery-background"
>
  <div
    className="lottery-background-1"
  >
    <div
      className="select-container"
    >
      <ForwardRef
        id="lottery-selector"
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "mega-sena",
              "value": 1,
            },
            Object {
              "label": "quina",
              "value": 2,
            },
          ]
        }
        styles={
          Object {
            "container": [Function],
            "control": [Function],
            "indicatorSeparator": [Function],
            "menu": [Function],
          }
        }
        value={
          Object {
            "label": "quina",
            "value": 2,
          }
        }
      />
    </div>
    <div
      className="contest-info-container"
    >
      <div
        className="lottery-info"
      >
        <img
          alt="logo-loteria"
          src="Logo_Sena.png"
        />
        <p />
      </div>
      <div
        className="contest-info"
      >
        <p>
          Concurso
        </p>
        <p>
          456
           - 
        </p>
      </div>
    </div>
  </div>
</div>
`;
